src/creature.cpp

```diff
@@ -66,14 +66,6 @@ static std::map<int, std::map<body_part, double> > default_hit_weights = {
     }
 };
 
-struct weight_compare {
```

Comment:
In general it is, if you mean "why doesn't either gcc or clang warn on this?", probably because there's some bizarre corner case in the standard that allows the definition to be used from a separate compilation unit or library, so it can't "prove" it's unused.
I'd be very surprised if there isn't a tool in the clang suite that will find these for you though.